### DISCUSSION TITLE
Writing a constraint for same shift repeatedly.

### DISCUSSION THREAD

**Discussion by liamnamba** [google/or-tools#2456] [CP-SAT questions] [answered]
I am working on a scheduler for a hospital that assigns residents to different wards. Each resident will work at the same ward 8 weeks in a row. I'm having trouble writing this constraint, what's the best way to do this?

This is what I have so far:
```
    min_shifts_at_same_ward = 8
    for r in all_residents:
        num_shifts_worked_same_ward = sum(shifts[(r,w,s,wa)]for w in all_weeks for s in all_shifts)
        model.Add(num_shifts_worked_same_ward == min_shifts_at_same_ward)
```

**Comment by lperron** [comment, marked as answer]
Late, but have you looked at the shift_scheduling_sat.py examples ?
